File: frontend/src/app/models/DTO/Project/projectCreateDTO.ts
/* Auto Generated */

import { Language } from "./../../Enums/language"
import { ProjectType } from "./../../Enums/projectType"
import { CompilerType } from "./../../Enums/compilerType"
import { AccessModifier } from "./../../Enums/accessModifier"

export interface ProjectCreateDTO {
    name: string;
    description: string;
    language: Language;
    projectType: ProjectType;
    compilerType: CompilerType;
    countOfSaveBuilds: number;
    countOfBuildAttempts: number;
    color: string;
    access: AccessModifier;
}

File: backend/IDE.Common/ModelsDTO/DTO/Project/ProjectCreateDTO.cs
using IDE.Common.Enums;

namespace IDE.Common.DTO.Project
{
    public class ProjectCreateDTO
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public Language Language { get; set; }
        public ProjectType ProjectType { get; set; }
        public CompilerType CompilerType { get; set; }
        public int CountOfSaveBuilds { get; set; }
        public int CountOfBuildAttempts { get; set; }
        public string Color { get; set; }
        public AccessModifier Access { get; set; }
    }
}
